sso/main.tf

```diff
@@ -0,0 +1,41 @@
+data "aws_caller_identity" "current" {}
+
+module "permission_sets" {
+  source   = "./modules/permission-sets"
+  for_each = { for item in var.associations : item.group => item }
+
+  permission_sets = [
+    {
+      name               = length(each.value.group) < 33 ? "${each.value.group}" : substr("${each.value.group}", 33, length(each.value.group))
+      policy_attachments = [each.value.policy]
+
+      tags             = lookup(each.value, "tags", {})
+      description      = lookup(each.value, "description", null)
+      relay_state      = lookup(each.value, "relay_state", "")
+      inline_policy    = lookup(each.value, "inline_policy", "")
+      session_duration = lookup(each.value, "session_duration", "")
+
+      customer_managed_policy_attachments = lookup(each.value, "customer_managed_policy_attachments", [])
+    }
+  ]
+}
+
+module "sso_account_assignments" {
+  source = "./modules/account-assignments"
+
+  for_each = { for item in var.associations : item.group => item }
+
+  account_assignments = [
+    {
+      account             = lookup(each.value, "account_id", var.account_id == "" ? data.aws_caller_identity.current.account_id : var.account_id),
+      permission_set_arn  = module.permission_sets["${each.value.group}"].permission_sets[length(each.value.group) <= 33 ? "${each.value.group}" : substr("${each.value.group}", 33, length(each.value.group))].arn,
+      permission_set_name = length(each.value.group) < 33 ? "${each.value.group}" : substr("${each.value.group}", 33, length(each.value.group))
+      principal_type      = "GROUP",
+      principal_name      = "test"
```

Comment:
is it ok to have "test" hardcoded here? if we can have anything here lets change this hardcoded value as having "test" in real live/prod setups can be ambiguous